source/common/upstream/load_balancer_impl.cc

```diff
@@ -52,19 +52,31 @@ std::pair<int32_t, size_t> distributeLoad(PriorityLoad& per_priority_load,
 
 } // namespace
 
-uint32_t LoadBalancerBase::choosePriority(uint64_t hash,
-                                          const std::vector<uint32_t>& per_priority_load) {
+std::pair<uint32_t, bool>
+LoadBalancerBase::choosePriority(uint64_t hash, const std::vector<uint32_t>& per_priority_load,
+                                 const std::vector<uint32_t>& degraded_per_priority_load) {
   hash = hash % 100 + 1; // 1-100
   uint32_t aggregate_percentage_load = 0;
   // As with tryChooseLocalLocalityHosts, this can be refactored for efficiency
   // but O(N) is good enough for now given the expected number of priorities is
   // small.
+
+  // We first attempt to select a priority based on healthy hosts.
   for (size_t priority = 0; priority < per_priority_load.size(); ++priority) {
     aggregate_percentage_load += per_priority_load[priority];
     if (hash <= aggregate_percentage_load) {
-      return priority;
+      return {priority, false};
+    }
+  }
+
+  // If no priorities were selected due to health, attempt to select a degraded one.
```

Comment:
"attempt to select a degraded one." This must succeed, right? Per the not reached below. Update the comment?